### PR TITLE
chore(preview-link): Make link to preview environment linkable

### PR DESCRIPTION
**Pull request by andromaqui**
## Description

This pr aims to make the link in the comment which notifies a user for the deployment of a preview environment clickable. I researched and found out that for this to work, the invisible tag has to be put at the end of the comment otherwise the markdown formatting breaks and the link shows as plan string.

## Test

I am attaching here screenshots of the messages.

On **deployment**:
Screenshot 2024-07-05 at 13 40 05
Screenshot 2024-07-05 at 14 19 02
Screenshot 2024-07-05 at 14 00 24

On **teardown**:
Screenshot 2024-07-05 at 14 26 10
Screenshot 2024-07-05 at 14 28 19

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: .github/workflows/preview-env-deploy.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-deploy.yml:91]:
```suggestion
            The preview environment relating to the commit ${{ env.short_sha }} has successfully been deployed. You can access it at [https://${{ env.BUCKET_NAME }}/pr-${{ github.event.number }}/index.html](https://${{ env.BUCKET_NAME }}/pr-${{ github.event.number }}/index.html) <!-- preview-env -->
```

**Review by maxdanilov — COMMENTED** [review]
LGTM overall, small grammar fix suggestion.

File: .github/workflows/preview-env-deploy.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-deploy.yml:32]: Short commit by GitHub standards is 7 chars not 8 ([ref](https://support.configura.com/hc/en-us/articles/360052644053-Are-Git-Commit-id-s-Truncated-in-CET-Developer#:~:text=The%20conclusion%20is%20that%20git,to%20always%20use%208%20characters.)):

```suggestion
        run: echo "short_sha=$(echo ${{ github.sha }} | cut -c1-7)" >> $GITHUB_ENV
```

File: .github/workflows/preview-env-deploy.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-deploy.yml:50]: Small suggestion for providing more context to the users

```suggestion
            :construction: The preview environment for the commit ${{ env.short_sha }} is being built. This usually takes 10-15 minutes. <!-- preview-env -->
```

File: .github/workflows/preview-env-deploy.yml
Comment (maxdanilov) [reply .github/workflows/preview-env-deploy.yml:32]: Actually, you also don't need to be doing the short SHA calculation at all. GitHub will shorten it automatically.

For example, when you mention in markdown
```
cfd70262bef268b3a615decc553e2208c18b29f5
```

it looks like this in the UI: cfd70262bef268b3a615decc553e2208c18b29f5

File: .github/workflows/preview-env-deploy.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-deploy.yml:83]: Small suggestion for providing more context to the users

```suggestion
            :arrow_up: The preview environment build results for commit ${{ env.short_sha }} is being uploaded. This usually takes 2-3 minutes. <!-- preview-env -->
```

Comment (maxdanilov) [reply .github/workflows/preview-env-deploy.yml:83]: (I've removed the `bucket` mention because I don't want to expose implementation details here, it's an open source project; contributors can figure this out when looking in the code, but we should not be openly mentioning it in the comment - same as hiding the bucket name in the secret)

File: .github/workflows/preview-env-teardown.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-teardown.yml:43]: same as above, you don't need to calculate short SHA.

File: .github/workflows/preview-env-teardown.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-teardown.yml:51]: Small suggestion for providing more context to the users
```suggestion
            :warning: Preview environment for commit ${{ env.short_sha }} is being deleted. <!-- preview-env -->
```

File: .github/workflows/preview-env-teardown.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-teardown.yml:75]: Small suggestion for providing more context to the users
```suggestion
            :broom: Preview environment for this PR has been torn down. <!-- preview-env -->
```

File: .github/workflows/preview-env-deploy.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-deploy.yml:109]:
```suggestion
            :link: The preview environment for commit ${{ env.short_sha }} has successfully been deployed. You can access it at [https://${{ env.BUCKET_NAME }}/pr-${{ github.event.number }}/index.html](https://${{ env.BUCKET_NAME }}/pr-${{ github.event.number }}/index.html) <!-- preview-env -->
```

File: .github/workflows/preview-env-deploy.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-deploy.yml:45]: this job needs to be moved before the `npm ci` step (this is where the build actually starts, with dep install)

File: .github/workflows/preview-env-teardown.yml
Comment (maxdanilov) [inline .github/workflows/preview-env-teardown.yml:48]: this step needs to be moved above, before the `Set up Cloud SDK` step, as that one takes quite some time (the idea is to notify the user of the deleting process as early as possible, here we're waiting for the GCloud to be installed and that can take a minute)

**Review by maxdanilov — DISMISSED** [review]
Thank you for the improvements, it looks better know with multiple different comments in the lifecycle.

I've also added suggestion for relocating the comment steps differently to have quicker update loop cycle for the user experience, and the suggestion to drop the short SHA calculation as unneeded.

File: .github/workflows/preview-env-deploy.yml
Comment (andromaqui) [reply .github/workflows/preview-env-deploy.yml:83]: Yeah this is actually a really good point

**Review by cmur2 — APPROVED** [review]
Requested changes got addressed afais, LGTM

**Comment by andromaqui** [comment]
@cmur2 thank you for the review. I think in the end I cant merge this PR until Maxim approves though, so I will wait until next week

**Comment by cmur2** [comment]
> @cmur2 thank you for the review. I think in the end I cant merge this PR until Maxim approves though, so I will wait until next week

Problem solved (should not the usual way to do it, but in this case it is fine):

image

**Comment by cmur2** [comment]
@andromaqui ^

**Comment by andromaqui** [comment]
Amazing, thank you!

**Pull request merged**
